src/genconfig.sh

```diff
@@ -1,12 +1,22 @@
 #!/bin/bash
 # SPDX-License-Identifier: GPL-2.0-only
 
+# ./genconfig.sh - generate C header for setting the correct preprocessor definitions for the kernel version
+# Usage ./genconfig.sh <kernel version> <make flags>
+# The make flags are used for determining concurrency for the feature tests, it pulls out the value of the -j flag.
+# ./genconfig.sh `uname -r` "-j4", four threads for running feature tests
+# ./genconfig.sh `uname -r` "-i -j4 -d", doesn't care about other flags present
+
 SRC_DIR=$(dirname "$0")
 OUTPUT_FILE=$SRC_DIR/kernel-config.h
 FEATURE_TEST_DIR="$SRC_DIR/configure-tests/feature-tests"
 FEATURE_TEST_FILES="$FEATURE_TEST_DIR/*.c"
 SYMBOL_TESTS_FILE="$SRC_DIR/configure-tests/symbol-tests"
 KERNEL_VERSION=$(uname -r)
+MAX_THREADS=$(echo "$2" | sed 's/.*-j\s*\([0-9]\+\).*/\1/')
+if [ -z "$MAX_THREADS" ]; then # if there was no -j flag provided, default to 1
+	MAX_THREADS=1
```

Comment:
Shouldn't we default to detecting instead by `/usr/bin/getconf _NPROCESSORS_ONLN` with a fallback of 1 if it's zero, and add an upper bound for the concurrency number (perhaps 8)?

Reply:
I don't think an upper bound is necessary, since setting `-j` to more than the number of cores doesn't break anything. I did change the default though.